prometheus-to-sd/translator/prometheus.go

```diff
@@ -62,7 +63,28 @@ func getPrometheusMetrics(config *config.SourceConfig) (*PrometheusResponse, err
 	if resp.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("request failed - %q, response: %q", resp.Status, string(body))
 	}
-	return &PrometheusResponse{rawResponse: string(body)}, nil
+	// TODO: Filter based on config.WhitelistedLabelsMap here
```

Comment:
This function should be responsible only for scraping /metrics endpoint.

Good place for filtering might be after we check whitelisted metrics: https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/prometheus-to-sd/translator/translator.go#L86

If you add that logic there it's going to be much simpler, because we have already parsed metric, and most of the hacks can be removed.

Reply:
And please don't forget to add tests once you move logic to translator.go

Reply:
Thanks for the review! I'll work on these changes.